### PR TITLE
Update collector flag initialization

### PR DESCRIPTION
**Pull request by bryan-aguilar**
**Description:** This PR reorders the way collector flag initialziation is performed. Instead of relying on the cobra command or windows service function to parse flags we parse flags manually once. This allows us to move config provider initialization into the main function. This also allows us to use feature gates when constructing default components. 

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: cmd/awscollector/main.go [hunk 1]
```diff
@@ -68,50 +66,73 @@ func main() {
...
+		if err := os.Setenv(awsProfileKey, extraCfg.AwsProfile); err != nil {
+			log.Printf("failed to set env var %s:%v\n", awsProfileKey, err)
+		}
 	}
+
 	if extraCfg.AwsCredentialFile != "" {
-		os.Setenv(awsCredentialFileKey, extraCfg.AwsCredentialFile)
+		if err := os.Setenv(awsCredentialFileKey, extraCfg.AwsCredentialFile); err != nil {
+			log.Printf("failed to set env var %s:%v\n", awsCredentialFileKey, err)
+		}
```
Comment (bryan-aguilar) [inline cmd/awscollector/main.go:125]: These changes are not related to feature gates but I decided to add some additional logging while I was making changes in this function.

File: cmd/awscollector/main.go [hunk 2]
```diff
@@ -68,50 +66,73 @@ func main() {
...
 		logFatal(err)
 	}
 }
 
+// Parse all the flags manually. We parse the flags manually here so that we can use feature gates when constructing
+// our default component list. Flags also need to be parsed before creating the config provider.
+func buildAndParseFlagSet(fs *flag.FlagSet, featReg *featuregate.Registry) error {
+	fs = config.Flags(featReg)
+
+	if err := flagSet.Parse(os.Args[1:]); err != nil {
```
Comment (Aneurysm9) [inline cmd/awscollector/main.go:96]:
```suggestion
	if err := fs.Parse(os.Args[1:]); err != nil {
```

**Review by erichsueh3 — APPROVED** [review]

File: cmd/awscollector/main.go [hunk 3]
```diff
@@ -39,6 +40,8 @@ const (
 	awsCredentialFileKey = "AWS_SHARED_CREDENTIALS_FILE" //nolint:gosec // this is a false positive for G101: Potential hardcoded credentials
 )
 
+var flagSet *flag.FlagSet
```
[the remark this answers is not in the document]
Comment (bryan-aguilar) [reply cmd/awscollector/main.go]: I removed the package global

File: cmd/awscollector/main.go (hunk 2 above)
Comment (bryan-aguilar) [reply cmd/awscollector/main.go:95]: This is out of date right?

**Review by Aneurysm9 — APPROVED** [review]

**Pull request merged**